posts/2023-02-27-introducing-new-conda.md

```diff
@@ -0,0 +1,53 @@
+---
+layout: post
+title: Introducing the new hh5 conda environment
+categories: python
+---
+# Introducing the new hh5 conda environment
+
+## Background
+The hh5 python/conda environment has grown vastly in scope since it was first installed many years ago. What started as a place to put important python packages missing from NCI's installations has since become an agile, community-driven, monolithic installation serving almost 600 researchers. Currently, there are 261 packages listed in the file used to construct and update the environment. Once all of the dependencies of these packages are factored in, there are 1,015 total python packages installed in the current `analysis3-unstable` environment. As a result of the work put in by my predecessors at CLEX CMS, an update to the `analysis3-unstable` environment, from a researcher requesting a package through cws_help@nci.org.au to integration and testing of that package, can be completed in under 1 hour.
+
+That being said, there are issues that had to be addressed going forward. The most important of these issues comes from the sheer size of the environment, at 1,015 packages, the current `analysis3-unstable` environment contains 289,524 files, directories and symlinks. At approximately 10GB in size, this corresponds to an average file size of around 36kB, which is not suitable for the lustre file systems that the conda environments are installed on. Secondly the lack of integration with NCI systems is a cause of problems. As conda is designed to create integrated environments on single-user systems, it installs packages like MPI and SSH, which are not configured correctly for NCI's systems, causing issues such as MPI not working across multiple nodes, or ssh host-based authentication failing.

```

Comment:
possibly make this paragraph a list of issues, just to break the big block of text